src/testflinger_device_connectors/devices/oemrecovery/oemrecovery.py

```diff
@@ -118,14 +184,21 @@ def check_device_booted(self):
         """Check to see if the device is booted and reachable with ssh"""
         logger.info("Checking to see if the device is available.")
         started = time.time()
+
         # Wait for provisioning to complete - can take a very long time
         while time.time() - started < 3600:
             try:
                 time.sleep(90)
-                self.copy_ssh_id()
-                return True
+                if (
+                    self.check_generic_classic()
+                    or self.check_device_initialized()
+                ):
+                    self.copy_ssh_id()
```

Comment:
almost! Move the self.copy_ssh_id() up *above* the check_... methods. That way you've already ensured that we can ssh to the device and that the key has been copied.

Then you can go back to those methods and remove all the sshpass stuff since we can just connect with the key now.

Reply:
sorry Paul, we can not move the copy ssh above the check. The issue Jerry encountered was copy ssh key before device is ready then found once device is ready the key is gone. So we have to make sure device initialed before we start copying ssh key or doing anything.

I tested it on two of my devices, I will try it on more other devices

Reply:
I understand now, thanks for the good description. I'm not likely to remember that we need this specific order though. Could we add a comment here that explains that if we copy the ssh key to the device before these checks are done, that it might disappear?

Otherwise, it looks good to me. We might want to land the big monorepo change first though. So we might have to rework this patch a bit to get it to work after that. It's small though, so I'm not too worried.